Library/Homebrew/livecheck/strategy/crate.rb

```diff
@@ -0,0 +1,115 @@
+        DEFAULT_REGEX = /^v?(\d+(?:\.\d+)+)$/i
+
+        # The default `strategy` block used to extract version information when
+        # a `strategy` block isn't provided.
+        DEFAULT_BLOCK = proc do |json, regex|
+          json["versions"]&.map do |version|
+            next if version["yanked"] == true
+
+            match = version["num"]&.match(regex)
+            next if match.blank?
+
+            match[1]
+          end
+        end.freeze
+
+        # The `Regexp` used to determine if the strategy applies to the URL.
+        URL_MATCH_REGEX = %r{
+          ^https?://static\.crates\.io/crates
+          /(?<package>[^/]+) # The name of the package
+          /.+\.crate # The crate filename
+        }ix
+
+        # Whether the strategy can be applied to the provided URL.
+        #
+        # @param url [String] the URL to match against
+        # @return [Boolean]
+        sig { params(url: String).returns(T::Boolean) }
+        def self.match?(url)
+          URL_MATCH_REGEX.match?(url)
+        end
+
+        # Extracts information from a provided URL and uses it to generate
+        # various input values used by the strategy to check for new versions.
+        #
+        # @param url [String] the URL used to generate values
+        # @return [Hash]
+        sig { params(url: String).returns(T::Hash[Symbol, T.untyped]) }
+        def self.generate_input_values(url)
+          values = {}
+
+          match = url.match(URL_MATCH_REGEX)
+          return values if match.blank?
+
+          values[:url] = "https://crates.io/api/v1/crates/#{match[:package]}/versions"
+
+          values
+        end
+
+        # Generates a URL and checks the content at the URL for new versions
+        # using {Json#versions_from_content}.
+        #
+        # @param url [String] the URL of the content to check
+        # @param regex [Regexp, nil] a regex for matching versions in content
+        # @param provided_content [String, nil] content to check instead of
+        #   fetching
+        # @param homebrew_curl [Boolean] whether to use brewed curl with the URL
+        # @return [Hash]
+        sig {
+          params(
+            url:              String,
+            regex:            T.nilable(Regexp),
+            provided_content: T.nilable(String),
+            homebrew_curl:    T::Boolean,
+            _unused:          T.nilable(T::Hash[Symbol, T.untyped]),
+            block:            T.nilable(Proc),
+          ).returns(T::Hash[Symbol, T.untyped])
+        }
+        def self.find_versions(url:, regex: nil, provided_content: nil, homebrew_curl: false, **_unused, &block)
+          match_data = { matches: {}, regex: regex, url: url }
+          match_data[:cached] = true if provided_content.is_a?(String)
+
+          generated = generate_input_values(url)
+          return match_data if generated.blank?
+
+          match_data[:url] = generated[:url]
+
+          content = if provided_content.is_a?(String)
```

Comment:
```suggestion
          content = if provided_content
```
Since `provided_content` is a `T.nilable(String)`, you should be able to simply this (also on line 91).